tests/testthat/test.2_purityA.R

```diff
@@ -66,35 +96,72 @@ test_that("checking frag4feature (fillpeaks)", {
+  # msPths[2] <- file.path(msPths[2], 'LCMS_2.mzML')
+  # msPths[3] <- file.path(msPths[3], 'LCMSMS_1.mzML')
+  # msPths[4] <- file.path(msPths[4], 'LCMSMS_2.mzML')
+  # ms_data = readMSData(msPths, mode = 'onDisk', msLevel. = 1)
+  #
+  # ##### subset the data to focus on retention times 30-90 seconds and m/z values between 100 and 200 m/z.
+  # rtr = c(30, 90)
+  # mzr = c(100, 200)
+  # ms_data = ms_data %>%  filterRt(rt = rtr) %>%  filterMz(mz = mzr)
+  #
+  # ##### perform feature detection in individual files
+  # cwp <- CentWaveParam(snthresh = 3, noise = 100, ppm = 10, peakwidth = c(3, 30))
+  # xcmsObj <- xcms::findChromPeaks(ms_data, param = cwp)
+  # xcmsObj@phenoData@data$class = c('blank', 'blank', 'sample', 'sample')
+  # xcmsObj@phenoData@varMetadata = data.frame('labelDescription' = 'sampleNames', 'class')
+  # pdp <- PeakDensityParam(sampleGroups = xcmsObj@phenoData@data$class, minFraction = 0, bw = 5, binSize = 0.017)
+  # xcmsObj <- groupChromPeaks(xcmsObj, param = pdp)
   msmsPths <- list.files(system.file("extdata", "lcms", "mzML", package="msPurityData"), full.names = TRUE, pattern = "MSMS")
 
-  pa <- readRDS(system.file("extdata", "tests", "purityA", "1_purityA_pa.rds", package="msPurity"))
-  xset <- readRDS(system.file("extdata","tests", "xcms", "msms_only_xset.rds", package="msPurity"))
-  pa@fileList[1] <- msmsPths[basename(msmsPths)=="LCMSMS_1.mzML"]
-  pa@fileList[2] <- msmsPths[basename(msmsPths)=="LCMSMS_2.mzML"]
-  xset@filepaths[1] <- msmsPths[basename(msmsPths)=="LCMSMS_1.mzML"]
-  xset@filepaths[2] <- msmsPths[basename(msmsPths)=="LCMSMS_2.mzML"]
-
-  xset <- xcms::fillPeaks(xset)
-
-  pa <- frag4feature(pa, xset)
-
-  #saveRDS(pa, file.path("inst", "extdata", "test_data", "purityA", "2_frag4feature_pa.rds"))
-  expect_equal(round(pa@grped_df$inPurity[1],4), 1)
-  expect_equal(round(pa@grped_df$precurMtchPPM[1], 4), 1.0048)
-  expect_equal(length(pa@grped_ms2), 77)
-  expect_equal(nrow(pa@grped_ms2[[2]][[1]]), 4)
-  expect_equal(round(pa@grped_ms2[[1]][[1]][1],4), 112.0509)
-
-  pa_saved <- readRDS(system.file("extdata", "tests", "purityA", "2_frag4feature_pa.rds", package="msPurity"))
-  expect_equal(pa@grped_ms2, pa_saved@grped_ms2)
-  expect_equal(pa@grped_df, pa_saved@grped_df)
-
+  fns = c(system.file("extdata","tests", "xcms", "msms_only_xcmsnexp.rds", package="msPurity"),
+          system.file("extdata","tests", "xcms", "msms_only_xset.rds", package="msPurity"))
+
+  for(fn in fns){
+    pa <- readRDS(system.file("extdata", "tests", "purityA", "1_purityA_pa.rds", package="msPurity"))
+    pa@fileList[1] <- msmsPths[basename(msmsPths)=="LCMSMS_1.mzML"]
+    pa@fileList[2] <- msmsPths[basename(msmsPths)=="LCMSMS_2.mzML"]
+    xcmsObj <- readRDS(fn)
+
+    if('XCMSnExp' == class(xcmsObj)[1]){
+      #xcmsObj <- xcms::fillChromPeaks(xcmsObj, param = ChromPeakAreaParam())
+      xcmsObj <- xcms::fillChromPeaks(xcmsObj, param = FillChromPeaksParam(expandMz = 0, expandRt = 0, ppm = 0))
```

Comment:
need to add the equivalent of 

```
      xcmsObj@filepaths[1] <- msmsPths[basename(msmsPths)=="LCMSMS_1.mzML"]
      xcmsObj@filepaths[2] <- msmsPths[basename(msmsPths)=="LCMSMS_2.mzML"]
```
for the new xcms objects as the filepaths for the saved object are hardcoded in msms_only_xcmsnexp.rds

Reply:
fixed